theurgy.lic

```diff
@@ -55,75 +130,357 @@ def buy_wine
 end
+
+invoke_parchment = proc do
+  DRCT.walk_to('1230')
+  if 'but there is nothing in there like that' != DRC.bput("rummage /C parchment my #{$settings.theurgy_supply_container}", 'but there is nothing in there like that', 'pale golden prayer parchment')
+    get_from_container('golden parchment')
+    fput 'invoke my parchment'
+    waitfor 'You conclude'
+    put_in_container('golden parchment')
+  end
+end
+
+######################## CUSTOM TRAVEL ########################
+
+enter_altar = proc do
+  DRCT.walk_to('1164')
+  move 'go shack'
+  move 'go hole'
+  move 'west'
+  move 'up'
+  waitrt?
+  move 'go path'
+end
+
+exit_altar = proc do
+  move 'go path'
+  move 'go hole'
+  waitrt?
+  move 'east'
+  move 'go hole'
+  move 'out'
+end
+
+######################## MISC ########################
+
+def research_check
+  case find_current_project = DRC.bput("research status", / (\d+)% complete with a portion .*\./, 'not researching anything', 'project about')
+  when /not researching anything/
+    return
+  when /project about/
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    $hold_research = research_type.find{ |$hold_research| find_current_project =~ /#{$hold_research}/i }
+    if $hold_research.nil?
+      return
+    end
+  when /complete with a portion/
+    echo find_current_project
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    $hold_research = research_type.find{ |$hold_research| find_current_project =~ /#{$hold_research}/i }
+    if $hold_research.nil?
+      return
+    end
+    if find_current_project[1,2].to_i >= 60
+      wait_time = (101 - find_current_project[1,2].to_i) * 3
+      Flags.reset('research_done')
+      pause (wait_time)
+    else
+      echo "stopping research"
+    end
+    fput 'prep bless'
+  end
+  echo $hold_research
+  return $hold_research
+end
+
+def controller(method)
+  echo Flags['theurgy-commune']
+  echo Flags['theurgy-eluned']
+  echo Flags['theurgy-tasmine']
+  echo Flags['theurgy-truffenyi']
+  echo Flags['theurgy-kertigen']
+  if Flags['theurgy-commune']
+    Flags.reset('theurgy-commune')
+    next_commune
+  end
+  method.call
+end
+
+def next_commune
+  if Flags['theurgy-eluned'] == false && DRStats.circle > 3 && DRSkill.getrank('Outdoorsmanship') >= 20 && DRSkill.getxp('Theurgy') <=31
+    comm_eluned
+  elsif Flags['theurgy-tasmine'] == false && DRStats.circle > 2 && DRSkill.getxp('Theurgy') <=31
+    comm_tasmine
+  elsif Flags['theurgy-truffenyi'] == false && DRStats.circle > 60 && DRSkill.getxp('Theurgy') <=24
+    comm_truff
+  elsif Flags['theurgy-kertigen'] == false && DRStats.circle > 8 && DRSkill.getxp('Theurgy') <=31
+    comm_kert
+  end
+end
+
+######################## FLAGS ########################
+  
+Flags.add('theurgy-commune', 'fully prepared to seek assistance from the Immortals once again', 'completed this commune too recently')
+Flags['theurgy-commune'] = true
```

Comment:
Yeah - I needed the functionality of the flag but I also want it to start true not false because it turns whenever it sees you are ready for a new commune (but at the start you are already ready)